### PR TITLE
pam-tests.py: add distribution specific test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add distribution specific test cases.
With the support of yaml, can run distribution and upstream test cases.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : 14087c50a7df391e0a3b85170adb2283f6ee821d
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-26T03.34-14087c5/job.log
 (1/2) pam-tests.py:PAM.test;run-run_type-upstream-1015: PASS (21.78 s)
 (2/2) pam-tests.py:PAM.test;run-run_type-distro-9c3f: ERROR: pam source install failed. (4.75 s)
RESULTS    : PASS 1 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-26T03.34-14087c5/results.html
JOB TIME   : 34.34 s


SLES:
JOB ID     : a73895f21f4b0591b5ce1185531fa7e1e73ce119
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-26T14.02-a73895f/job.log
 (1/2) pam-tests.py:PAM.test;run-run_type-upstream-1015: PASS (19.73 s)
 (2/2) pam-tests.py:PAM.test;run-run_type-distro-9c3f: PASS (39.66 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-26T14.02-a73895f/results.html
JOB TIME   : 77.88 s
```

**Comment by nasastry** [comment]
```
SLES:
JOB ID     : b982acb8fc58dd37036caf2b46923fd196b06f5c
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T13.49-b982acb/job.log
 (1/2) pam-tests.py:PAM.test;run-run_type-upstream-1015: FAIL: pam-tests.py: 'make check' failed. (23.41 s)
 (2/2) pam-tests.py:PAM.test;run-run_type-distro-9c3f: PASS (46.36 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T13.49-b982acb/results.html
JOB TIME   : 91.65 s


RHEL:
JOB ID     : afbda2e6624cbb89299f6371cda38150b4e9cb23
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T03.22-afbda2e/job.log
 (1/2) pam-tests.py:PAM.test;run-run_type-upstream-1015: FAIL: pam-tests.py: 'make check' failed. (25.20 s)
 (2/2) pam-tests.py:PAM.test;run-run_type-distro-9c3f: ERROR: pam source install failed. (4.59 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T03.22-afbda2e/results.html
JOB TIME   : 42.74 s
```

File: security/pam-tests.py [hunk 1]
```diff
@@ -35,25 +35,42 @@ def setUp(self):
...
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://github.com/linux-pam/linux-pam/archive/master.zip"
-        tarball = self.fetch_asset(url, expire='7d')
-        archive.extract(tarball, self.workdir)
-        self.sourcedir = os.path.join(self.workdir, 'linux-pam-master')
-        os.chdir(self.sourcedir)
-        process.run('./autogen.sh', ignore_status=True)
-        process.run('./configure', ignore_status=True)
+        run_type = self.params.get('type', default='upstream')
```
Comment (Naresh-ibm) [inline security/pam-tests.py:38]: @nasastry change default='distro'

File: security/pam-tests.py [hunk 2]
```diff
@@ -35,25 +35,42 @@ def setUp(self):
...
+        if run_type == "upstream":
+            default_url = ("https://github.com/linux-pam/linux-pam/"
+                           "archive/master.zip")
+            url = self.params.get('url', default=default_url)
```
Comment (Naresh-ibm) [inline security/pam-tests.py:42]: @nasastry remove default_url variable and change default=None

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Review by Naresh-ibm — APPROVED** [review]
@nasastry @abdhaleegit Thanks for all efforts 
LGTM

**Pull request merged**